Add SQA requirements

aeslaughter opened the pull request:
<!--
INCLUDE THE FOLLOWING IN THE PR DESCRIPTION
- Explain relevant design information for your change.
- Follow the [Coding Standards](http://mooseframework.org/wiki/CodeStandards/).
- Submit or improve [Test Cases](http://mooseframework.org/wiki/MooseTraining/testing/).
- Reference a specific issue, place "refs #<issue>" or "closes #<issue>" (e.g., #closes #1234).
-->

aeslaughter committed:
Add req doco for 'tests/executioners'

(refs #9638)

aeslaughter committed:
Req doco for 'tests/fixedbugs'

(refs #9638)

aeslaughter committed:
req doco for 'tests/functions'

(refs #9638)

Commit:
Add requirement to nullspace tests

Text was provided by Sebastian Schunert

(refs #9638)

moosebuild commented:
Job [Documentation](https://civet.inl.gov/job/227883/) on 3113f6b wanted to post the following:

View the site [here](http://mooseframework.inl.gov/docs/PRs/12086/site)

This comment will be updated on new commits.

aeslaughter committed:
req doco for 'tests/geomsearch'

(refs #9638)

aeslaughter commented:
```
Requirement Definitions (26.3% complete):
                 Location: /Users/slauae/projects/moose/test/tests
                    Specs: tests
    Total Number of Tests: 1784
  Tests with Requirements: 470
```

aeslaughter committed:
Add requirements to modified test files

(refs #9638)

aeslaughter committed:
The SQA check must check that a file exists

(refs #12082)
The git command will return files that have been moved or deleted, so check first

permcody commented on test/tests/geomsearch/1d_line_sampler/tests:
```diff
@@ -1,7 +0,0 @@
-  [./test]
-    type = 'CSVDiff'
-    input = '1d_line_sampler.i'
-    csvdiff = '1d_line_sampler_out_my_vpp_0001.csv'
```
If you delete this test, you also need to delete the gold files and input files.

permcody replied:
Oh I see, it's a rename... NVM.
